mesh/mesh.hpp

```diff
@@ -579,8 +579,27 @@ class Mesh
        new mesh. Periodic meshes are not supported. */
    static Mesh MakeSimplicial(const Mesh &orig_mesh);
 
+   /// Create a periodic mesh by identifying vertices of @a orig_mesh.
+   /** Each vertex @a i will be mapped to vertex @a v2v[i], such that all
+       vertices that are coincident under the periodic mapping get mapped to
+       the same index. The mapping @a v2v can be generated from translation
+       vectors using Mesh::CreatePeriodicVertexMapping. */
+   static Mesh MakePeriodic(const Mesh &orig_mesh, const std::vector<int> &v2v);
+
    ///@}
 
+   /// @brief Creates a mapping @a v2v from the vertex indices of the mesh such
+   /// that coincident vertices under the given @a translations are identified.
+   /** Each Vector in @a translations should be of size @a sdim (the spatial
+       dimension of the mesh). Two vertices are considered coincident if the
+       translated coordinates of one vertex are within the given tolerance (@a
+       tol, relative to the mesh diameter) of the coordinates of the other
+       vertex.
+       @warning This algorithm does not scale well with the number of vertices
+       in the mesh, and is not recommended for very large meshes.  */
+   std::vector<int> CreatePeriodicVertexMapping(
```

Comment:
At least for the return value, `Array<int>` should definitely be possible. Avoiding an extraneous copy of the vertex map in cases like
```c++
Mesh::MakePeriodic(mesh, mesh.CreatePeriodicVertexMapping(translations));
```
would then rely on (non-mandatory) NRVO, whereas `std::vector` has move semantics, so it is mandated by the standard.

On the other hand, for the translations, we can't use `Array<Vector>` because `Vector` is not a trivial type (cf. #2214). I would really prefer to avoid `Array<Vector*>`.

Reply:
So, I'm a fan of using `std::vector<T>` here based on what I've done on another project that used the `Mesh::MakePeriodic()`. It made creating the v2v vector fairly simple, since I was able to build up off a number of `std::algorithms`.